app/src/main/java/com/cornellappdev/scoop/ui/components/general/NavHeader.kt

```diff
@@ -0,0 +1,110 @@
+package com.cornellappdev.scoop.ui.screens
+
+import androidx.compose.foundation.BorderStroke
+import androidx.compose.foundation.Canvas
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.PathEffect
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.navigation.NavController
+import com.cornellappdev.scoop.R
+import com.cornellappdev.scoop.ui.theme.Green
+import com.google.accompanist.pager.PagerState
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+
+@OptIn(ExperimentalMaterialApi::class, com.google.accompanist.pager.ExperimentalPagerApi::class)
+@Composable
+fun NavHeader(
+    title: String,
+    backFunction: suspend () -> Unit,
+    hasBackArrow: Boolean = false,
+) {
+    val scope = rememberCoroutineScope()
+
+    Column(
+        Modifier.fillMaxWidth()
+            .padding(0.dp, 20.dp, 0.dp, 0.dp),
+    ) {
+        Row(
+            Modifier.fillMaxWidth(),
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.Center,
+        ) {
+
+            if(hasBackArrow){
+                Button(
+                    onClick = {
+                        scope.launch {
+                            withContext(Dispatchers.IO){
```

Comment:
Does this work without the withContext? I think the scope.launch should be enough to run the suspend function backFunction()

Reply:
They aren't needed, just tested. Pushed change